src/oid.rs

```diff
@@ -78,14 +78,20 @@ pub struct ObjectId {
     id: [u8; 12],
 }
 
+impl Default for ObjectId {
```

Comment:
Clippy suggests that we implement `Default` for types when they have a constructor with the singature `fn new() -> Self`. While it seems a little weird to have a `Default` implementation that doesn't always implement the same thing, the [documentation for `Default`](https://doc.rust-lang.org/std/default/trait.Default.html) indicates that the default value can be "anything else that may make sense as a default", and it doesn't say anything about invariants that the value must always be the same. I think this would be useful for users who put ObjectIds in structs; if they want to derive `Default`, they almost definitely don't want an ObjectId with all `0` bytes.